### PR TITLE
Add Echo/SQL Sample

### PR DESCRIPTION
**Pull request by petergeorgas**
This branch creates the exact same URL shortening service in the gin-mongo example, but using Echo and PostgreSQL.

Instructions are largely the same as far as running this sample service, with mainly just one additional step of spinning up the PostgreSQL docker container. The container will be configured automatically upon first creation. 

Takes care of #3

### CODE REVIEW

File: echo-sql/README.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+ go test -coverpkg=./... -covermode=atomic  ./...
+```
+
+output should look like
+
+```shell
+ok      test-app-url-shortener  6.268s  coverage: 80.3% of statements in ./...
+```
+
+**We got 80.3% without writing any testcases or mocks for mongo db!!**
```
Comment (slayerjain) [inline echo-sql/README.md:88]: we should change this line

File: echo-sql/README.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+Now that we have our testcase captured, run the test file.
+
+```shell
+ go test -coverpkg=./... -covermode=atomic  ./...
+```
+
+output should look like
+
+```shell
+ok      test-app-url-shortener  6.268s  coverage: 80.3% of statements in ./...
```
Comment (slayerjain) [inline echo-sql/README.md:85]: We should update this line to reflect the actual code coverage we are getting

File: echo-sql/README.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+```shell
+git clone https://github.com/keploy/keploy.git && cd keploy
+docker-compose up
+```
+
+### Setup URL shortener
+
+```bash
+git clone https://github.com/keploy/example-url-shortener && cd example-url-shortener/echo-sql
```
Comment (slayerjain) [inline echo-sql/README.md:17]: please update the URL, since it's changed now. We also need to update this for the other gin-mongo example.

**Review by slayerjain — CHANGES_REQUESTED** [review]
also, if i shutdown the Postgres database and run tests it fails with the following error: 

Screenshot 2022-03-15 at 9 33 47 AM

@re-Tick let's find out why this is happening.

File: echo-sql/README.md (hunk 3 above)
Comment (petergeorgas) [reply echo-sql/README.md]: Fixed in most recent commit.

**Comment by petergeorgas** [comment]
@slayerjain Any update on the Postgres shutdown issue? Anything I can do?

**Comment by slayerjain** [comment]
@petergeorgas we found another problem with error deserialization. @re-Tick has identified a fix thanks to the amazing go community - https://www.reddit.com/r/golang/comments/tgfvyp/expression_comparing_two_equal_errors_is/

I think we should be able to merge and release within a day :)

**Comment by slayerjain** [comment]
@petergeorgas We've released the fix. please update the SDK version to `v0.3.5`and record test cases and run test mode again. Now you should be able to successfully run the test cases even after shutting down Postgres 👯

**Comment by petergeorgas** [comment]
@slayerjain All should be good to go. I updated to the most recent Keploy SDK 0.3.5, and updated the README to reflect actual test results. Working really well!

**Comment by slayerjain** [comment]
@petergeorgas that's awesome. Checking this out :)

**Comment by slayerjain** [comment]
Thanks @petergeorgas  👯

File: echo-sql/README.md (hunk 2 above)
Comment (petergeorgas) [reply echo-sql/README.md]: Code coverage has now been updated.

**Pull request merged**